.github/workflows/ui-tests/cypress/integration/deposit_spec.js

```diff
@@ -0,0 +1,192 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const defaultUser = Cypress.env('defaultUser')
+const username = defaultUser.username
+const password = defaultUser.password
+const name = defaultUser.name
+const externalAccount = defaultUser.externalAccounts[0]
+
+const depositMsgs = Cypress.env('messages').deposit
+const invalidFeedback = Cypress.env('messages').invalidFeedback
+const formatter = new Intl.NumberFormat('en-US', {
+    maximumFractionDigits: 2,
+  });
+
+const randomInt = (min, max) => {
+    // max is exclusive and the min inclusive
+    return Math.floor(Math.random() * (max-min)) + min
+}
+
+const validPayment = () => {
+    const max = 100
+    const min = 1
+    const num = (Math.random() * max) + min
+    return formatter.format(num)
+}
+
+const validAccountNum = () => {
+    // 10 digit integer
+    const max = 10000000000
+    const min = 1000000000
+    return randomInt(min, max)
+}
+
+const validRoutingNum = () => {
+    // 9 digit integer
+    const max = 1000000000
+    const min = 100000000
+    return randomInt(min, max)
+}
+
+describe('Authenticated default user', function () {
+    beforeEach(function () {
+        cy.loginRequest(username, password)
+        cy.visit('/home')
+    })
+
+    it('sees deposit button', function () {
+        cy.get('#depositSpan').contains('Deposit Funds')
+    })
+
+    it('clicking deposit button makes modal visible', function () {
+        cy.get('#depositFunds').should('not.be.visible')
+        cy.get('#depositSpan').click()
+        cy.get('#depositFunds').should('be.visible')
+    })
+
+    it('sees expected external accounts', function () {
+        cy.get('#depositSpan').click()
+        cy.get('#depositFunds').should('be.visible')
+        cy.get('#accounts').children().first().as('firstOption')
+        cy.get('@firstOption').contains('External Bank')
+        cy.get('@firstOption').contains('9099791699')
+        cy.get('@firstOption').contains('808889588')
+    })
+
+    it('can deposit funds successfully', function () {
+        const depositAmount = validPayment()
+
+        cy.deposit(externalAccount, depositAmount)
+        cy.get('.alert').contains(depositMsgs.success)
+    })
+
+    it('can see balance update after deposit', function () {
+        const depositAmount = validPayment()
+        let expectedBalance
+        cy.get("#current-balance").then(($span) => {
+            const currentBalanceSpan = $span.text()
+            // regex: removes any characters that are not a digit [0-9] or a period [.]
+            const currentBalance = parseFloat(currentBalanceSpan.replace(/[^\d.]/g, ''))
```

Comment:
maybe we should only remove , and $ if those are all we expect to see?

Reply:
when cypress retrieves the span there are newline characters that had to be removed as well